runtime/near-vm-runner/src/cache.rs

```diff
@@ -250,14 +255,26 @@ impl FilesystemContractRuntimeCache {
     ///
     /// Note though, that this memory cache is *not* used to additionally cache files from the
     /// filesystem – OS page cache already does that for us transparently.
-    pub fn with_memory_cache<SP: AsRef<std::path::Path> + ?Sized>(
+    pub fn with_memory_cache<SP, CCP>(
         home_dir: &std::path::Path,
         store_path: Option<&SP>,
+        contract_cache_path: &CCP,
         memory_cache_size: usize,
-    ) -> std::io::Result<Self> {
+    ) -> std::io::Result<Self>
+    where
+        SP: AsRef<std::path::Path> + ?Sized,
+        CCP: AsRef<std::path::Path> + ?Sized,
+    {
         let store_path = store_path.map(AsRef::as_ref).unwrap_or_else(|| "data".as_ref());
-        let path: std::path::PathBuf =
+        let legacy_path: std::path::PathBuf =
             [home_dir, store_path, "contracts".as_ref()].into_iter().collect();
+        let path: std::path::PathBuf =
+            [home_dir, contract_cache_path.as_ref()].into_iter().collect();
+        // Rename the old contracts directory to a new name. This should only succeed the first
+        // time this code encounters the legacy contract directory. If this fails the first time
+        // for some reason, a new directory will be created for the new cache anyway, and future
+        // launches won't be able to overwrite it anymore. This is also fine.
+        let _ = std::fs::rename(legacy_path, &path);
```

Comment:
It works I guess but I would feel better with something more explicit like:
* if legacy exists && not current exists -> log, move, report result
* if legacy exists && current exists -> log, remove legacy, report result

Reply:
Would it make sense to check for existence of the legacy directory *after* the attempted rename, and if it's present advise that it was deprecated, won't be used, and can be deleted?

Reply:
Added a warning to this effect.